kong/router/compat.lua

```diff
@@ -92,8 +157,34 @@ local function get_expression(route)
   if gen then
     -- See #6425, if `net.protocol` is not `https`
     -- then SNI matching should simply not be considered
-    gen = "(net.protocol != \"https\"" .. LOGICAL_OR .. gen .. ")"
+    gen = "((net.protocol != \"https\" && net.protocol != \"tls\")" .. LOGICAL_OR .. gen .. ")"
```

Comment:
This makes an extra unneccesary condition for both system. We should not sacrifice runtime performance(when executing the route).
```suggestion
    gen = "(net.protocol != \"" .. (is_http and "https" or "tls") .. "\"" .. LOGICAL_OR .. gen .. ")"
```

Reply:
I'm not fammilar with this part of code. Maybe the detection should be done by checking the route dynamically. Correct me if I'm wrong.

Reply:
This function should generate both http and stream expression, because we can only manage expression by http endpoint in http subsystem.

Reply:
To make it more clear, I don't think it's necessary to check if the protocol is "tls" for the HTTP system, and vice versa.

Reply:
done. thanks.